tests/integration-tests/conftest.py

```diff
@@ -1022,6 +1065,78 @@ def _create_bucket():
                 logging.error(f"Failed deleting bucket {bucket[0]} with exception: {e}")
 
 
+@xdist_session_fixture(autouse=True)
+def s3_buckets(session_s3_bucket_factory, request):
+    regions = request.config.getoption("regions") or get_all_regions(request.config.getoption("tests_config"))
+    s3_buckets_dict = {}
+    for region in regions:
+        s3_bucket = session_s3_bucket_factory(region)
+        scheduler_definitions_dict = _upload_scheduler_plugin_definitions(request, s3_bucket, region)
```

Comment:
we could transform `upload_scheduler_plugin_definitions` into a xdist_session_fixture that depends on s3_buckets just to keep the two logic separate

Reply:
done